frontend/test/unit/contract.test.js

```diff
@@ -0,0 +1,172 @@
+import { connect } from "get-starknet";
+import axios from "axios";
+import {
+  deployContract,
+  checkAndDeployContract,
+} from "../../src/utils/contract";
+import { getDeployContractData } from "../../src/utils/constants";
+
+jest.mock("get-starknet");
+jest.mock("axios");
+jest.mock("../../src/utils/constants");
+
+describe("Contract Deployment Tests", () => {
+  const mockWalletId = "0x123...";
+  const mockTransactionHash = "0xabc...";
+  const mockContractAddress = "0xdef...";
+  const mockBackendUrl = "http://127.0.0.1:8000";
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    process.env.REACT_APP_BACKEND_URL = mockBackendUrl;
+
+    getDeployContractData.mockReturnValue({
+      contractData: "mockContractData",
+    });
+  });
+
+  describe("deployContract", () => {
+    test("should successfully deploy contract", async () => {
```

Comment:
stick to the single approach across all the test files and use either `it` or `test` keyword everywhere. I'd suggest/prefer `it`